source/geometries/Next100FieldCage.cc

```diff
@@ -757,17 +786,20 @@ void Next100FieldCage::BuildFieldCage()
   G4double first_ring_drift_z_pos = posz - ring_drift_buffer_dist_;
   for (G4int i=0; i<num_drift_rings_; i++) {
     posz = first_ring_drift_z_pos - i*drift_ring_dist_;
-    new G4PVPlacement(0, G4ThreeVector(0., 0., posz),
+    new G4PVPlacement(0, G4ThreeVector(GetCoordOrigin().x(),
+                                       GetCoordOrigin().y(), posz),
                       ring_logic, "FIELD_RING", mother_logic_,
                       false, i, false);
   }
 
   // Ring vertex generator
   G4double ring_gen_lenght = first_ring_buff_z_pos + ring_thickn_/2. - (posz - ring_thickn_/2.);
   G4double ring_gen_zpos = first_ring_buff_z_pos + ring_thickn_/2. - ring_gen_lenght/2.;
-  ring_gen_ = new CylinderPointSampler2020(ring_int_diam_/2., ring_ext_diam_/2., ring_gen_lenght/2.,
-                                           0., twopi, nullptr,
-                                           G4ThreeVector(0., 0., ring_gen_zpos));
+  ring_gen_ =
+    new CylinderPointSampler2020(ring_int_diam_/2., ring_ext_diam_/2., ring_gen_lenght/2.,
```

Comment:
This is not related to the PR but I just realized that `ring_gen_lenght` has a typo. And it might be spreaded throughout the whole code.

Reply:
Fixed!